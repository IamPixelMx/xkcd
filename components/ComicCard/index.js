import ComicCard from './ComicCard';

export { ComicCard };
