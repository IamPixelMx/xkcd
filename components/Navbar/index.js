import Navbar from './Navbar';
import NavbarInput from './NavbarInput';
import NavbarItem from './NavbarItem';

export { Navbar, NavbarInput, NavbarItem };
