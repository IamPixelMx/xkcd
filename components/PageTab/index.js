import PageTab from './PageTab';

export { PageTab };
