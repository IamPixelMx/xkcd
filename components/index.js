import { Footer } from './Footer';
import { Navbar } from './Navbar';
import { PageTab } from './PageTab';

export { Footer, Navbar, PageTab };
