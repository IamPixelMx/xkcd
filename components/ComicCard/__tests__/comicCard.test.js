import React from 'react';
import configureStore from 'redux-mock-store';
import { fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import { renderWithProvider, renderWithProviderSnapshot } from 'jest-utils';
import { comicData } from 'jest-factories';
import { ComicCard } from 'components';

const mockStore = configureStore([]);

const toogleFavItem = jest.fn().mockImplementation(props => {
  return props ? 'delete' : 'add';
});

const comicCardElement = <ComicCard comicData={comicData} toogleFavItem={toogleFavItem} />;

describe('ComicCard', () => {
  it('Renders ComicCard ', () => {
    const comicCard = renderWithProviderSnapshot(comicCardElement, {
      mockStore,
    }).toJSON();
    expect(comicCard).toMatchSnapshot();
  });

  it('Call Toogle function with button card', async () => {
    const storeWithFav = mockStore({
      favorites: [comicData],
    });

    const { getByText } = renderWithProvider(comicCardElement, {
      storeWithFav,
    });
    const cardButton = getByText('Agregar a Favoritos ❤');
    expect(cardButton).toBeInTheDocument();
    await act(async () => {
      fireEvent.click(cardButton);
    });
    expect(toogleFavItem).toBeCalled();
  });
});
