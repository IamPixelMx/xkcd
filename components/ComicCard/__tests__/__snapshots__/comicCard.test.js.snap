// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicCard Renders ComicCard  1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <span>
      <p
        className="title"
      >
        Exposure Notification
      </p>
    </span>
    <time
      dateTime="5-8-2020"
    >
      <small>
        5 de septiembre de 2020
      </small>
    </time>
  </div>
  <div
    className="card-image"
  >
    <figure
      className="image comic"
    >
      <img
        alt="comic-image"
        src="https://imgs.xkcd.com/comics/exposure_notification.png"
      />
    </figure>
  </div>
  <div
    className="card-content has-text-centered mb-2"
  >
    <p
      className="subtitle"
    >
      I don't see why everyone is so hungry for BAD news, but fine, I'll give in to feedback and add a dark mode.
    </p>
  </div>
  <footer
    className="card-footer"
  >
    <p
      className="card-footer-item"
    >
      <a
        onClick={[Function]}
      >
        Agregar a Favoritos ‚ù§
      </a>
    </p>
  </footer>
</div>
`;
